debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerConnectorTask.java

```diff
@@ -0,0 +1,200 @@
+
+    private volatile SqlServerTaskContext taskContext;
+    private volatile ChangeEventQueue<Object> queue;
+    private volatile SqlServerConnection jdbcConnection;
+    private volatile ChangeEventSourceCoordinator coordinator;
+    private volatile ErrorHandler errorHandler;
+    private volatile SqlServerDatabaseSchema schema;
+    private volatile Map<String, ?> lastOffset;
+
+    @Override
+    public String version() {
+        return Module.version();
+    }
+
+    @Override
+    public void start(Configuration config) {
+        if (!state.compareAndSet(State.STOPPED, State.RUNNING)) {
+            LOGGER.info("Connector has already been started");
+            return;
+        }
+
+        final SqlServerConnectorConfig connectorConfig = new SqlServerConnectorConfig(config);
+        taskContext = new SqlServerTaskContext(connectorConfig);
+
+        final Clock clock = Clock.system();
+
+        // Set up the task record queue ...
+        this.queue = new ChangeEventQueue.Builder<Object>()
+                .pollInterval(connectorConfig.getPollInterval())
+                .maxBatchSize(connectorConfig.getMaxBatchSize())
+                .maxQueueSize(connectorConfig.getMaxQueueSize())
+                .loggingContextSupplier(() -> taskContext.configureLoggingContext(CONTEXT_NAME))
+                .build();
+
+        errorHandler = new ErrorHandler(SqlServerConnector.class, connectorConfig.getLogicalName(), queue, this::cleanupResources);
+        final SqlServerTopicSelector topicSelector = SqlServerTopicSelector.defaultSelector(connectorConfig.getLogicalName());
+
+        final Configuration jdbcConfig = config.subset("database.", true);
+
+        jdbcConnection = new SqlServerConnection(jdbcConfig, new SqlServerConnectionFactory());
+        final SchemaNameAdjuster schemaNameAdjuster = SchemaNameAdjuster.create(LOGGER);
+
+        this.schema = new SqlServerDatabaseSchema(connectorConfig, schemaNameAdjuster, topicSelector, jdbcConnection);
+
+        final SqlServerOffsetContext previousOffset = null;
+//        OracleOffsetContext previousOffset = getPreviousOffset(connectorConfig);
+//        if (previousOffset != null) {
+//            schema.recover(previousOffset);
+//        }
+
+        final EventDispatcher<TableId> dispatcher = new EventDispatcher<>(topicSelector, schema, queue,
+                connectorConfig.getTableFilters().dataCollectionFilter(), DataChangeEvent::new);
+
+        coordinator = new ChangeEventSourceCoordinator(
+                previousOffset,
+                errorHandler,
+                SqlServerConnector.class,
+                connectorConfig.getLogicalName(),
+                new SqlServerChangeEventSourceFactory(connectorConfig, jdbcConnection, errorHandler, dispatcher, clock, schema)
+        );
+
+        coordinator.start();
+    }
+
+//    private OracleOffsetContext getPreviousOffset(SqlServerConnectorConfig connectorConfig) {
```

Comment:
That part already got a bit simpler with `BaseSourceTask#getPreviousOffset(Loader)`.